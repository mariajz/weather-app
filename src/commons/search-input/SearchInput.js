import React from 'react';
import { TextInput } from 'react-native';
import { SearchInputWrapper } from './SearchInput.style';
import { SearchIcon } from '../visual-elements';

const SearchInput = ({ placeholder }) => {
    return (
        <SearchInputWrapper>
            <TextInput
                placeholder={placeholder}
                placeholderTextColor={'black'}
            />
            <SearchIcon />
        </SearchInputWrapper>
    );
};

export default SearchInput;
