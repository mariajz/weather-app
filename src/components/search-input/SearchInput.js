import React, { useState } from 'react';
import { View } from 'react-native';
import { Divider } from '../../commons/styles';
import { CustomIcon } from '../../commons/visual-elements';
import {
    DropDown,
    ModalWrapper,
    Row,
    SearchInputWrapper,
    StyledCustomIcon,
    StyledIcon,
    StyledModal,
    StyledText,
    StyledTextInput,
    StyledTouchableOpacity,
} from './SearchInput.style';

const SearchInput = ({ placeholder }) => {
    const [showDropDown, setShowDropDown] = useState(false);
    const [showModal, setShowModal] = useState(true);

    const [searchIconPressed, setSearchIconPressed] = useState(false);

    const [locations] = useState([
        { name: 'abc', country: 'India', key: 1 },
        { name: 'abc', country: 'India', key: 2 },
        { name: 'abc', country: 'India', key: 3 },
        { name: 'abc', country: 'India', key: 4 },
        { name: 'abc', country: 'India', key: 5 },
    ]);

    const handleOnSearchIconPress = () => {
        setSearchIconPressed(!searchIconPressed);
        setShowModal(true);
    };

    const handleOnChangeText = data => {
        if (data.length === 0) {
            setShowDropDown(false);
        }
        if (data.length > 3) {
            // make api call to fetch locations
            if (locations.length !== 0) {
                setShowDropDown(true);
            } else {
                setShowDropDown(false);
            }
        }
    };

    const handleOnDropDownItemPress = () => {
        // make api call to fetch forecast data
    };

    return (
        <SearchInputWrapper>
            <If condition={searchIconPressed}>
                <ModalWrapper>
                    <StyledTouchableOpacity
                        onPress={() => {
                            setShowModal(true);
                        }}>
                        <View>
                            <StyledModal
                                visible={showModal}
                                onBackdropPress={() => {
                                    setShowDropDown(false);
                                    setShowModal(false);
                                    setSearchIconPressed(!searchIconPressed);
                                }}>
                                <StyledTextInput
                                    placeholder={placeholder}
                                    placeholderTextColor="black"
                                    onChangeText={handleOnChangeText}
                                />
                                <If condition={showDropDown}>
                                    <DropDown>
                                        <If condition={locations.length > 0}>
                                            {locations.map(
                                                (location, index) => {
                                                    return (
                                                        <React.Fragment
                                                            key={index}>
                                                            <Row
                                                                onPress={() => {
                                                                    handleOnDropDownItemPress(
                                                                        location,
                                                                    );
                                                                }}>
                                                                <CustomIcon
                                                                    iconName="pin"
                                                                    size={12}
                                                                    tintColor="black"
                                                                />
                                                                <StyledText>
                                                                    {
                                                                        location?.name
                                                                    }
                                                                    ,
                                                                    {
                                                                        location?.country
                                                                    }
                                                                </StyledText>
                                                            </Row>
                                                            <Divider />
                                                        </React.Fragment>
                                                    );
                                                },
                                            )}
                                        </If>
                                    </DropDown>
                                </If>
                            </StyledModal>
                        </View>
                    </StyledTouchableOpacity>
                </ModalWrapper>
            </If>
            <StyledIcon>
                <If condition={searchIconPressed}>
                    <StyledCustomIcon
                        iconName="close"
                        onPress={handleOnSearchIconPress}
                        tintColor="black"
                        size="14"
                    />
                    <Else />
                    <CustomIcon
                        testID="search-icon"
                        iconName="search"
                        onPress={handleOnSearchIconPress}
                        tintColor="black"
                    />
                </If>
            </StyledIcon>
        </SearchInputWrapper>
    );
};

export default SearchInput;
