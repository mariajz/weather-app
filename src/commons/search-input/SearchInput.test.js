import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import SearchInput from './SearchInput';

jest.mock('../visual-elements', () => ({
    SearchIcon: () => <MockView />,
    Theme: {
        BackgroundWhite: jest.fn(
            opacity => `mocked rgba(255, 255, 255, ${opacity})`,
        ),
    },
}));

describe('SearchInput', () => {
    it('should render SearchInput', () => {
        const container = render(<SearchInput />);

        expect(container).toMatchSnapshot();
    });
});
