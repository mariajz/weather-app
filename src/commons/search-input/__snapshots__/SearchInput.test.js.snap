// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput should render SearchInput 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "mocked rgba(255, 255, 255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "space-between",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 10,
      "paddingBottom": 0,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 0,
    }
  }
>
  <TextInput
    placeholderTextColor="black"
  />
  <View />
</View>
`;
